views/forms.twig

```diff
@@ -393,3 +393,65 @@
         </div>
     </div>
 {% endmacro %}
+
+{% macro dateRangeFilter(name, title, value, helpText, groupClass, validation, accessKey) %}
+    <div class="form-group mr-1 mb-1 d-flex flex-row {{ groupClass }}">
+        {% set today =  now | date_modify('today') | date("Y-m-d H:i:s") %}
+        <div class="form-group mr-1">
+            <label class="control-label mr-1" title="{{ helpText }}" for="{{ name }}" accesskey="{{ accessKey }}">
+                {{ title }}
+            </label>
+            <div class="d-inline-flex">
+                <select class="form-control XiboDateRangeFilter" name="{{ name }}" id="{{ name }}">
+                    <option value="" >{% trans "Select a range" %}</option>
+                    <option value="today" selected>{% trans "Today" %}</option>
+                    <option value="yesterday">{% trans "Yesterday" %}</option>
+                    <option value="thisweek">{% trans "This Week" %}</option>
+                    <option value="thismonth">{% trans "This Month" %}</option>
+                    <option value="thisyear">{% trans "This Year" %}</option>
+                    <option value="lastweek">{% trans "Last Week" %}</option>
+                    <option value="lastmonth">{% trans "Last Month" %}</option>
+                    <option value="lastyear">{% trans "Last Year" %}</option>
+                </select>
+            </div>
+        </div>
+        <div class="form-group hidden mr-1 rangeFilterInput">
+            <label class="control-label mr-1" title="{{ helpText }}" for="{{ name }}" accesskey="{{ accessKey }}">
+                {% trans "From Date" %}
+            </label>
+            <div class="input-group">
+                <div class="input-group-prepend input-group-text date-open-button" role="button">
+                    <i class="fa fa-calendar"></i>
+                </div>
+                <input class="form-control dateControl date rangeInput"
+                       type="text" name="fromDt" id="fromDt"
```

Comment:
Great, this reusable control is much better! Do we need to think about making the IDs here be specific to this control? If we include this macro more than once we will currently have duplicated IDs. The name would need to stay the same to it matches the API, but the ID could be unique to this instance (perhaps we could make the ID fromDt_name?)

Reply:
Hi, Dan! Thank you for this feedback! Please see updated changes [here](https://github.com/xibosignage/xibo-cms/pull/2827/commits/ec9e3dad2a216a20738eb41dab70a5513515fa50) :)